Allow prompt_ret_code xontrib to set background color

m-col opened the pull request:
I discovered xonsh last week and I've been loving it.

This change extends the prompt_ret_code xontrib. It looks for `$PROMPT_RET_CODE_BACKGROUND` and if it is `True` then the return code will set the prompt background color, rather than foreground color.

I wasn't sure how to look for the existence of a variable nicely and ending up using a `try` block. Is this the best way?

This PR also fixes a typo in xonsh/aliases.py.

gforsyth commented on xontrib/prompt_ret_code.xsh:
```diff
@@ -2,20 +2,31 @@ from xonsh.tools import ON_WINDOWS as _ON_WINDOWS
 
 def _ret_code_color():
+    try:
+	color_bg = $PROMPT_RET_CODE_BACKGROUND
```
For xonsh env-vars, we try to start them with `XONSH_` in an effort to namespace them and keep things a bit organized -- this is a xontrib, although a bundled xontrib.  What do you think about prefixing it with `XONTRIB_`?

m-col replied:
Seems reasonable.

gforsyth commented on xontrib/prompt_ret_code.xsh:
```diff
@@ -2,20 +2,31 @@ from xonsh.tools import ON_WINDOWS as _ON_WINDOWS
+	color_bg = False
+
+    if color_bg:
+	prefix = 'BACKGROUND'
```
Another option here is to allow the env-var to take the values of `BOLD` or `BACKGROUND` (and potentially others) and then use the value directly.

Take a look in `xonsh.environ` at the `DEFAULT_ENSURERS` for how we type and validate env-var inputs

You can create an ensurer for the acceptable value types and then add that ensurer using

```
${...}.set_ensurer(env_var_name, ensurer_func)
```

m-col replied:
Good idea. My proposed changes extend hard coded colors to slightly more hard coded colors, but the functionality should probably be flexible enough to have any color string that would be valid in `$PROMPT_FIELDS`. This would only require a variable for each of zero or non-zero return codes.

I'll have a go setting up an ensurer for these two variables to make sure they are valid color strings.

m-col replied:
Actually - it would be 4 variables. zero and non-zero for background and foreground.

gforsyth commented on news/prompt_ret_code_background.rst:
```diff
@@ -0,0 +1,24 @@
+
+**Changed:**
+
+* added $PROMPT_RET_CODE_BACKGROUND which causes the prompt_ret_code xontrib to
```
```suggestion
* added ``$PROMPT_RET_CODE_BACKGROUND`` which causes the prompt_ret_code xontrib to
```

Just a small formatting thing, but adding double backticks makes the release notes look nicer

scopatz replied:
Also, this doesn't seem to be the name of the variable any longer.

m-col commented on xontrib/prompt_ret_code.xsh:
```diff
@@ -2,20 +2,31 @@ from xonsh.tools import ON_WINDOWS as _ON_WINDOWS
 
 def _ret_code_color():
+    try:
+	color_bg = $PROMPT_RET_CODE_BACKGROUND
```
OK - that is much nicer that the try block!

m-col commented:
I boiled down the settings for color format to two variables, one for each of zero and non-zero return codes:
```
$XONTRIB_PROMPT_RET_CODE_COLOR_ZERO
$XONTRIB_PROMPT_RET_CODE_COLOR_NONZERO
```
I added ensurers for these too to make sure they are strings. Please let me know if this wasn't done correctly!

I also moved the ```ON_WINDOWS``` test out of the prompt function so it's only tested when the xontrib is loaded rather than during every draw. I assume this was there originally due to different color codes on windows, and I figured simply setting the default for windows in this way would be fine.

The last thing to add is documentation (and update the news file). I read around in your link to environ.py to see how documentation is handled. I didn't manage to add documentation using an equivalent method - I suppose this is what you meant by the lack of "public-facing" env-var documentation setting. Is there a temporary workaround until such a functionality exists?

Of course please do let me know of any improvements to my commits!

laloch commented:
What's up with this PR? Did we reject, abandon or just forget it?

m-col commented:
I thought you'd forgotten about it as I didn't hear back from that last comment. I stopped using xonsh since then so happy to close it.
